packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,310 @@
+import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button'
+import KeyboardAwareScrollView from '@celo/react-components/components/KeyboardAwareScrollView'
+import KeyboardSpacer from '@celo/react-components/components/KeyboardSpacer'
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import variables from '@celo/react-components/styles/variables'
+import { parseInputAmount } from '@celo/utils/lib/parsing'
+import { RouteProp } from '@react-navigation/core'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import React, { useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { Platform, StyleSheet, Text, TextInput, View } from 'react-native'
+import { getNumberFormatSettings } from 'react-native-localize'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { cUsdDailyLimitSelector } from 'src/account/selectors'
+import BackButton from 'src/components/BackButton'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import Dialog from 'src/components/Dialog'
+import LineItemRow from 'src/components/LineItemRow'
+import { CELO_SUPPORT_EMAIL_ADDRESS, DOLLAR_ADD_FUNDS_MIN_AMOUNT } from 'src/config'
+import { fetchExchangeRate } from 'src/exchange/actions'
+import { exchangeRatePairSelector } from 'src/exchange/reducer'
+import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
+import i18n, { Namespaces } from 'src/i18n'
+import { LocalCurrencySymbol } from 'src/localCurrency/consts'
+import {
+  convertDollarsToLocalAmount,
+  convertDollarsToMaxSupportedPrecision,
+  convertLocalAmountToDollars,
+} from 'src/localCurrency/convert'
+import { useLocalCurrencyCode } from 'src/localCurrency/hooks'
+import { getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
+import { emptyHeader, HeaderTitleWithBalance } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import { getRateForMakerToken, goldToDollarAmount } from 'src/utils/currencyExchange'
+
+const { decimalSeparator } = getNumberFormatSettings()
+
+type RouteProps = StackScreenProps<StackParamList, Screens.FiatExchangeAmount>
+
+type Props = RouteProps
+
+const oneDollarAmount = {
+  value: new BigNumber('1'),
+  currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+}
+
+const useDollarAmount = (currency: CURRENCY_ENUM, amount: BigNumber) => {
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
```

Comment:
Can we have all of these as function params rather than calling then using the selectors again? Seems like we are already using all of these in the main component function so it would be more efficient to pass them as params rather than making additional calls to the Redux store.